zcash_client_backend/src/sync.rs

```diff
@@ -0,0 +1,399 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::anyhow;
+use futures_util::TryStreamExt;
+use prost::Message;
+use tokio::{fs::File, io::AsyncWriteExt, task::JoinHandle};
+use tonic::{
+    body::BoxBody,
+    client::GrpcService,
+    codegen::{Body, Bytes, StdError},
+};
+use tracing::{debug, error, info};
+// TODO: Move `FsBlockDb` behind a `BlockCache` trait to break cyclic dependency.
+use zcash_client_sqlite::{chain::BlockMeta, FsBlockDb, FsBlockDbError};
+use zcash_primitives::{
+    consensus::{BlockHeight, Parameters},
+    merkle_tree::HashSer,
+};
+
+use crate::{
+    data_api::{
+        chain::{error::Error as ChainError, scan_cached_blocks, BlockSource, CommitmentTreeRoot},
+        scanning::{ScanPriority, ScanRange},
+        WalletCommitmentTrees, WalletRead, WalletWrite,
+    },
+    proto::service::{self, compact_tx_streamer_client::CompactTxStreamerClient},
+};
+
+pub(crate) fn get_block_path(fsblockdb_root: &Path, meta: &BlockMeta) -> PathBuf {
+    meta.block_file_path(&fsblockdb_root.join("blocks"))
+}
+
+/// Scans the chain until the wallet is up-to-date.
+pub async fn run<P, ChT, DbT>(
```

Comment:
This method only scans CompactBlocks, which contain only shielded transaction data. Transparent data is totally omitted. I had to fill in the gap in my sync routine, which is mostly complete, although zcash_client_backend itself doesn't yet support recording a *spend* of a transparent output, so while I download transparent transactions, the recorded history is incorrect if UTXOs are spent, but it works if they are shielded. 
In any case, I think the transparent support either needs to be added here, or the function should be very prominently documented as to its limitations.